### DISCUSSION TITLE
How can I get the location of a node?

### DISCUSSION THREAD

**Discussion by 6peterlu** [facebook/lexical#2766] [Q&A]
I'm looking for a way to get the location of a node in `RangeSelection` format:
```
export type NodeLocation = {
  focus: { offset: number; type: string };
  anchor: { offset: number; type: string };
};
```
What's the best way to do that? In this case focus would be the start of the node and anchor would be the end.

Thank you!

**Comment by kulame** [comment]
I have the same problem, is there any good solution?

**Comment by acywatson** [comment]
I'm not sure I understand what you're asking for here.

So, focus is typically a Node and anchor is a node, and the offset specifies the position within the children or text content (for TextNodes). The position of an actual node can only really be discussed in terms of it's position in the tree relative to others nodes. For instance, it would have a parent and it might be the Nth child of that (parent) node.

Let me know if that doesn't make sense or maybe take a step back and tell me what you're ultimately trying to do.
